### ISSUE TITLE
[MobileDatePicker] Picker does not open on first click when value is null and clicking on the label

### ISSUE THREAD

**Issue by ori1989** [mui/mui-x#5395]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

MobileDatePicker does not open on the first TextField click when value is null, only after multiple clicks on TextField.

<MobileDatePicker
      label="my-date"                      
      inputFormat="DD/MM/YYYY"
      value={null}                                                  
      DialogProps={{ className: 'mui-datepicker' }}
      onChange={(date) => this.handleChange('my-date', date)}
      onOpen={this.handleOpen}
      renderInput={(params) => <TextField fullWidth {...params} />}
  />

### Expected behavior 🤔

It's supposed to open on first TextField click.

### Steps to reproduce 🕹

Steps:

1.
2.
3.
4.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by flaviendelangle** [comment]
Hi,
Thanks for your contribution.

I could not reproduce your problem 
Could you fork this [Codesandbox](https://codesandbox.io/s/date-and-time-pickers-forked-87t9j8?file=/src/App.tsx) to reproduce your issue ?

**Comment by flaviendelangle** [comment]
I would really be a lot easier with a code reproduction (via Codesandbox or another similar platform) rather than a video

**Comment by ori1989** [comment]
1. Go to the expermiment app url:
https://87t9j8.csb.app/
2. Go to Console Developer (F12):
3. Choose a small screen device (For example, IPhone SE)
4. Click on the MobileDatePicker TextField (click on a left side point - for example on placeholder label "my-date" when it's not focused). And You'll see in some cases it opens and in some cases it does not open.

P.S: It seems it does not open correctly only on small devices when I click on the left side of the TextField.

[MobileDatePicker - bug.webm](https://user-images.githubusercontent.com/34649863/177701398-cdb29dc3-bfe2-4831-93a3-9d4c0176c05a.webm)

**Comment by LukasTy** [comment]
After quite a bit of testing and debugging I think I've narrowed the bug down to these facts:
- it only happens when clicking on the label (when no value is present)
- it only happens in chrome (Firefox on Android works without problems)

Hence, we need the least intrusive fix to cover some sort of issue in chrome.

**Comment by troussema** [comment]
:+1:   I have the same problem on Firefox, Chrome and Safari on IOS :

https://user-images.githubusercontent.com/32413550/181765320-97677b87-0ff7-4891-95ed-b12f89961e93.mp4

**Comment by LukasTy** [comment]
@troussema Thank you for sharing this information.
Could you look into the [proposed solution](https://deploy-preview-5651--material-ui-x.netlify.app/x/react-date-pickers/date-picker/) and if it resolves the issue on iOS Safari?

Note: As far as I'm aware iOS does not allow different browsers ship their own engine implementation, hence every iOS browser app is just a skin for the native WebKit. ([source](https://infrequently.org/2022/06/apple-is-not-defending-browser-engine-choice/))

**Comment by troussema** [comment]
@LukasTy  :1st_place_medal: Thank you for your fast reply ! 
The fix work like a charm ! :clap:  ( tested on Chrome,Firefox and Safari ) 

https://user-images.githubusercontent.com/32413550/181773198-d512be38-ecb3-4a1d-a940-4622ca791f94.MP4

Am using : Iphone 13 pro max with IOS 15.6

**Comment by murillo94** [comment]
@LukasTy  I know this issue is closed, but I noticed the same behavior on MobileDateRangePicker from mui-pro.

You need to do a double-click to open the date range.

**Comment by flaviendelangle** [comment]
Hi,
Thanks for your contribution.

Could you provide a working reproduction, this [Codesandbox template](https://codesandbox.io/s/date-and-time-pickers-uj6btq) might be a good starting point.

**Comment by LukasTy** [comment]
@murillo94, could you please clarify what browser/environment and package version are we talking about?
I've tried `v5`, `v6`, and `7` on Safari (on an iOS simulator), Chrome on Android Phone, and mobile emulation on desktop Chrome. It seems to work correctly everywhere. 🤷

**Comment by nachtfunke** [comment]
> [@troussema](https://github.com/troussema) Thank you for sharing this information. Could you look into the [proposed solution](https://deploy-preview-5651--material-ui-x.netlify.app/x/react-date-pickers/date-picker/) and if it resolves the issue on iOS Safari?
> 
> Note: As far as I'm aware iOS does not allow different browsers ship their own engine implementation, hence every iOS browser app is just a skin for the native WebKit. ([source](https://infrequently.org/2022/06/apple-is-not-defending-browser-engine-choice/))

So I am running in the exact same issue, but your proposed solution just links to a documentation page. If there is a proposed solution in there, I cannot find it.

**Comment by LukasTy** [comment]
> So I am running in the exact same issue, but your proposed solution just links to a documentation page. If there is a proposed solution in there, I cannot find it.

@nachtfunke The link pointed to a PR deployment with the fix (https://github.com/mui/mui-x/pull/5651) applied for OP to check if it fixes the issue.
If you are having issues with this in a newer version (>=6), please consider opening a new issue.
